src/main/java/org/folio/template/service/TemplateServiceImpl.java

```diff
@@ -55,16 +58,19 @@ public TemplateServiceImpl(Vertx vertx, Map<String, String> okapiHeaders) {
   }
 
   public Future<List<Template>> getTemplates(String query, int offset, int limit) {
+    LOG.debug("getTemplates:: Retrieving Templates with query {}", query);
     return templateDao.getTemplates(query, offset, limit);
   }
 
   @Override
   public Future<Optional<Template>> getTemplateById(String id) {
+    LOG.debug("getTemplateById:: Retrieving Template by ID : {}", id);
     return templateDao.getTemplateById(id);
   }
 
   @Override
   public Future<String> addTemplate(Template template) {
+    LOG.debug("addTemplate:: Adding Template with ID : {}", template.getId());
     validateTemplate(template);
     if (template.getId() == null) {
       template.setId(UUID.randomUUID().toString());
```

Comment:
Please add log for the condition block of code it has been missed many places. Please check all the places and follow the logging guide lines of this improvement work.